### PR TITLE
refactor: update modals v5design

### PR DESCRIPTION
**Pull request by surekhaw**

### CODE REVIEW

File: content/modals/DE/product_list.json [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
-                    "viewName": "PAY_LATER_LONG_TERM"
-                }
-            ]
-        }
-    }
-}
+{
+    "meta": {
+        "product": "PRODUCT_LIST",
+        "useV5Design": "true"
```
Comment (yeul) [inline content/modals/DE/product_list.json]:
```suggestion
        "useV5Design": "false"
```

File: src/components/modal/v2/parts/Icon.jsx [hunk 2]
```diff
@@ -118,6 +118,23 @@ const Icon = ({ name = '', color, content }) => {
                     />
                 </svg>
             );
+        case 'v5Design':
```
Comment (yeul) [inline src/components/modal/v2/parts/Icon.jsx]: The logo between v4 and v5 low-scope is the same svg and size so we can remove this for now.

File: src/components/modal/v2/styles/globals/_colors.scss [hunk 3]
```diff
@@ -12,10 +12,10 @@ $offer-border-gray: #c6c6c6;
 $beige: #faf8f5;
 // Lander background
 $lander-beige: #e6e0d9;
-// v4 lander background
-$v4-lander-gray: #e9e9e9;
-// v4 header/footer background
-$v4-gray: #f6f6f6;
+// v5 background
```
Comment (yeul) [inline src/components/modal/v2/styles/globals/_colors.scss:21]: Let's add back the v4 colors for now because checkout is still going to be on V4 for the time being.

File: src/components/modal/v2/parts/Header.jsx [hunk 4]
```diff
@@ -40,24 +40,28 @@ const Header = ({
...
             <div className="header__icons">
-                <div className={`logo__wrapper ${isScrolled ? 'logo__wrapper--scroll' : ''}`}>
+                <div
+                    className={`logo__wrapper ${isScrolled ? 'logo__wrapper--scroll' : ''} ${
+                        useV5Design ? 'v5Design' : ''
+                    }`}
+                >
                     <div className="pp-logo">
-                        <Icon name={logo} />
+                        <Icon name={useV5Design ? 'v5Design' : logo} />
```
Comment (yeul) [inline src/components/modal/v2/parts/Header.jsx]: We can revert this change for now since the logos are the same between v4 and v5.

File: src/components/modal/v2/styles/components/_header.scss [hunk 5]
```diff
@@ -138,6 +138,12 @@
                 .pp-logo {
                     pointer-events: none;
                 }
+
+                &.v5Design {
+                    @include tablet {
+                        margin: 50px 56px 8px;
+                    }
+                }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_header.scss]: We're safe to remove this for now as the logos will look and behave the same.

File: src/components/modal/v2/styles/components/_header.scss [hunk 6]
```diff
@@ -184,6 +190,10 @@
                     stroke-width: 2.2;
                 }
             }
+
+            &.v5Design {
+                margin-bottom: 50px;
+            }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_header.scss]: Same as above.

File: src/components/modal/v2/styles/components/_header.scss [hunk 7]
```diff
@@ -374,6 +392,45 @@
...
+                @include tablet {
+                    margin-bottom: 10px;
+                    margin-left: 20px;
+                    padding: 0px 10px 0px 0px;
+                }
+                @include mobile {
+                    padding: 10px 22px 50px;
+                }
+            }
+        }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_header.scss]: Since we're keeping the content and logo the same, we can revert this change for now as well.

File: src/components/modal/v2/styles/components/_calculator.scss [hunk 8]
```diff
@@ -178,6 +186,18 @@
...
+            color: $v5-gray;
+        }
+    }
+
+    &.v5Design {
+        text-align: left;
+        align-items: left;
+        @include tablet {
+            width: 503px;
+        }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_calculator.scss]: Did we confirm if we want the US modal calc content full width for our desktop view? Or is it staying as it is now where the calc terms are in the middle of the modal with the larger padding on the sides?

File: content/modals/GB/short_term.json [hunk 9]
```diff
@@ -7,7 +7,6 @@
         "qualifying": "{qualifying_offer}",
         "amount": "{transaction_amount}",
         "apr": "{apr}",
-        "useV4Design": "false",
```
Comment (yeul) [inline content/modals/GB/short_term.json:10]: Did we mean to delete this completely from this file instead of updating to `V5`?

File: src/components/modal/v2/parts/OfferAccordion.jsx [hunk 10]
```diff
@@ -18,8 +25,13 @@ const OfferAccordion = ({ offer: { content, meta }, aprDisclaimer, index, active
     }, []);
 
     return (
-        <div id={index} className={`accordion__container ${activeSelection === index ? 'active' : ''} ${open}`}>
-            <div className="accordion__row">
+        <div
+            id={index}
+            className={`accordion__container ${activeSelection === index ? 'active' : ''} ${open} ${
+                useV5Design ? 'v5Design' : ''
```
Comment (yeul) [inline src/components/modal/v2/parts/OfferAccordion.jsx]:
```suggestion
                useV5Design === 'true' ? 'v5Design' : ''
```

File: src/components/modal/v2/parts/OfferAccordion.jsx [hunk 11]
```diff
@@ -18,8 +25,13 @@ const OfferAccordion = ({ offer: { content, meta }, aprDisclaimer, index, active
...
+            }`}
+        >
+            <div className={`accordion__row ${useV5Design ? 'v5Design' : ''}`}>
```
Comment (yeul) [inline src/components/modal/v2/parts/OfferAccordion.jsx]:
```suggestion
            <div className={`accordion__row ${useV5Design === 'true' ? 'v5Design' : ''}`}>
```

File: src/components/modal/v2/styles/components/_instructions.scss [hunk 12]
```diff
@@ -145,6 +151,10 @@
         }
     }
 
+    &.v5Design {
+        padding-top: 14px;
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_instructions.scss:155]:
```suggestion
        padding-top: 14px;
        margin-bottom: 50px;
```

File: src/components/modal/v2/styles/components/_content.scss [hunk 13]
```diff
@@ -137,6 +143,14 @@
                 margin-top: 10px;
             }
 
+            &.v5Design {
+                color: $v5-gray;
+                margin-top: 20px;
+                &.DE {
+                    margin-top: 80px;
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_content.scss]:
```suggestion
                    margin-top: auto;
```

File: src/components/modal/v2/styles/components/_offer-accordion.scss [hunk 14]
```diff
@@ -22,6 +22,25 @@
...
+                color: $v5-white;
+                button {
+                    background-color: #0070e0;
+                    color: $v5-white;
+                }
+                div {
+                    color: $v5-white !important;
+                }
+            }
+        }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_offer-accordion.scss:44]: We can add the below underneath your `&.open` styles to ensure the cards take up the desired width.
```suggestion
        }
        
          .v5Design & {
            width: 100%;

            &.open {
                border: 3px solid $light-blue;
            }
        }
```

File: src/components/modal/v2/styles/components/_calculator.scss [hunk 15]
```diff
@@ -110,6 +110,18 @@
             }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_calculator.scss:110]: Making sure the input field width matches the figma.
```suggestion
            }
            
             .v5Design & {
                width: 98%;
            }
```

File: src/components/modal/v2/styles/components/_calculator.scss [hunk 16]
```diff
@@ -110,6 +110,18 @@
...
+        &.v5Design {
+            h4 {
+                font-weight: 700;
+            }
+            label {
+                color: $v5-gray;
+            }
+            input {
+                border: 1px solid $v5-gray;
+            }
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_calculator.scss:140]:
```suggestion
            }
            
              #purchase-amount {
                width: 100%;
            }
```
Follow up from the above.

File: src/components/modal/v2/styles/components/_offer-accordion.scss [hunk 17]
```diff
@@ -22,6 +22,25 @@
             outline: 3px solid $light-blue;
             border-color: transparent;
         }
+        .v5Design {
+            button {
+                background-color: $ql-gray;
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_offer-accordion.scss]:
```suggestion
                background-color: $ql-gray;
                border-radius: 11px;
```

File: src/components/modal/v2/styles/components/_offer-accordion.scss
Comment (yeul) [inline src/components/modal/v2/styles/components/_offer-accordion.scss:1]: [Inside of this block](https://github.com/paypal/paypal-messaging-components/blob/2bacb0db183ba0e5dfbe3a2dad1cc72aaf33bc3f/src/components/modal/v2/styles/components/_offer-accordion.scss#L46-L57) can we add the following. It will ensure the amount/mo. is bolded:
```
 .accordion__offer-field-header {
            &:first-child {
                .v5Design & {
                    font-weight: bold;
                }
            }
        }
```

Comment (yeul) [inline src/components/modal/v2/styles/components/_offer-accordion.scss:1]: [Inside of this block](https://github.com/paypal/paypal-messaging-components/blob/2bacb0db183ba0e5dfbe3a2dad1cc72aaf33bc3f/src/components/modal/v2/styles/components/_offer-accordion.scss#L78-L80), can we also add to match the mock:
```
.v5Design & {
                        text-decoration: underline;
                    }
```

File: content/modals/GB/short_term.json (hunk 9 above)
Comment (surekhaw) [reply content/modals/GB/short_term.json:10]: yes. it's not listed in the -3308 short term v5 modal story, and it's not in the content repo. I deleted to match the content repo.

Comment (yeul) [reply content/modals/GB/short_term.json:10]: Thanks for the reminder!

File: src/components/modal/v2/styles/components/_tile.scss
Comment (yeul) [inline src/components/modal/v2/styles/components/_tile.scss:1]: [Under this line](https://github.com/paypal/paypal-messaging-components/blob/fa22f6ca4c76d87ad3df0a581d33d9337b7f3eeb/src/components/modal/v2/styles/components/_tile.scss#L22), can we add the following since it looks like there's no box shadow by default on the designs:
```
   &.v5Design {
        box-shadow: none;
    }
```
Adding it there will allow it to continue to exist on hover which is good for a user to see that it's a button.

File: src/components/modal/v2/parts/views/ShortTerm/styles.scss [hunk 18]
```diff
@@ -145,6 +145,11 @@
     }
 
     .disclosure {
+        &.v5Design {
+            color: $v5-gray;
+            padding-top: 24px;
```
Comment (yeul) [inline src/components/modal/v2/parts/views/ShortTerm/styles.scss:157]:
```suggestion
            padding-top: 24px;
            margin-top: auto !important;
```
Just to keep the disclosure pinned to the bottom.

File: src/components/modal/v2/parts/views/ShortTerm/styles.scss
Comment (yeul) [inline src/components/modal/v2/parts/views/ShortTerm/styles.scss:1]: [Under this line](https://github.com/paypal/paypal-messaging-components/blob/fa22f6ca4c76d87ad3df0a581d33d9337b7f3eeb/src/components/modal/v2/parts/views/ShortTerm/styles.scss#L46), can we add:
```
   .v5Design & {
                @include tablet {
                    margin-top: 20px;
                    margin-bottom: 20px;
                }
            }
```
To decrease the larger gap on tablet between the instructions, donuts, and disclosure. Just making things look more balanced.

**Review by yeul — COMMENTED** [review]
Can we also make a slight update to the `LoadingShimmer` file right above `&:first-child` [here](https://github.com/paypal/paypal-messaging-components/blob/fa22f6ca4c76d87ad3df0a581d33d9337b7f3eeb/src/components/modal/v2/styles/components/_loading-shimmer.scss#L70):

```
    &.shimmer {
            .v5Design & {
                width: 100%;
            }
        }
```
I noticed the loading shimmer wasn't full width on the US modals. This should fix it!

**Review by yeul — APPROVED** [review]

File: src/components/modal/v2/styles/components/_offer-accordion.scss [hunk 19]
```diff
@@ -22,6 +22,31 @@
...
+            .accordion__row {
+                button {
+                    border-radius: 11px;
+
+                }
+            }
+        }
+
+        &.open {
+            &.v5Design {
```
Comment (yeul) [inline src/components/modal/v2/styles/components/_offer-accordion.scss:39]:
```suggestion
            &.v5Design {
            outline: none;
```
Something I missed in my original review - I noticed a thin white outline around the offer boxes for DE and this should fix it.
